src/features.jl

```diff
@@ -0,0 +1,94 @@
+equalvolumeradius(radii) = sum(radii.^3)^(1/3)
+equalvolumeradius(atoms::AbstractVector{PDBAtom}, r::PDBResidue) = equalvolumeradius([vanderwaalsradius[(r.id.name, a.atom)] for a in atoms])
+
+function combinedfeaturesize(atoms, r)
+    μ = mean(a -> a.coordinates, atoms)
+    length(atoms) == 1 && return equalvolumeradius(atoms, r)
+    return std([norm(a.coordinates .- μ) for a in atoms]) + equalvolumeradius(atoms, r)
+end
+
+function add_feature!(mgmm::IsotropicMultiGMM, key, μ, σ, ϕ)
+    if !haskey(mgmm.gmms, key)
+        push!(mgmm.gmms, key => IsotropicGMM([IsotropicGaussian(vec(μ), σ, ϕ)]))
+    else
+        push!(mgmm.gmms[key].gaussians, IsotropicGaussian(vec(μ), σ, ϕ))
+    end
+end
+
+function add_features_from_atom!(mgmm::IsotropicMultiGMM, a::PDBAtom, r::PDBResidue, σfun, ϕfun)
+    σ = σfun(a, r)
+    ϕ = ϕfun(a, r)
+    if ishydrophobic(a, r.id.name)
+        add_feature!(mgmm, :Hydrophobe, a.coordinates, σ, ϕ)
+    end
+    if isaromatic(a, r.id.name)
+        add_feature!(mgmm, :Aromatic, a.coordinates, σ, ϕ)
+    end
+    if iscationic(a, r.id.name)
+        add_feature!(mgmm, :PosIonizable, a.coordinates, σ, ϕ)
+    end
+    if isanionic(a, r.id.name)
+        add_feature!(mgmm, :NegIonizable, a.coordinates, σ, ϕ)
+    end
+    if ishbonddonor(a, r.id.name)
+        add_feature!(mgmm, :Donor, a.coordinates, σ, ϕ)
+    end
+    if ishbondacceptor(a, r.id.name)
+        add_feature!(mgmm, :Acceptor, a.coordinates, σ, ϕ)
+    end
+end
+
+function add_features_from_atoms!(mgmm::IsotropicMultiGMM, key, atoms, r, σfun, ϕfun)
+    isempty(atoms) && return
+    μ = mean(a -> a.coordinates, atoms)
+    σ = σfun(atoms, r)
+    ϕ = ϕfun(atoms, r)
+    add_feature!(mgmm, key, μ, σ, ϕ)
+end
+
+function add_features_from_residue!(
+    mgmm::IsotropicMultiGMM, 
+    r::PDBResidue; 
+    combined = false, 
+    σfun = combined ? combinedfeaturesize : (a, r) -> vanderwaalsradius[(r.id.name, a.atom)],
+    ϕfun = combined ? (atoms, r) -> length(atoms) : (args...) -> 1.0
+)
```

Comment:
Based on your review [here](https://github.com/HolyLab/GPCRAnalysis.jl/pull/10#discussion_r1597128738), I've changed things a bit to let a user pass functions for determining `σ` and `ϕ` for a residue. By default, they should do both of the things you suggested earlier (use `length` for `ϕ` and account for the spatial distribution of atoms in `σ`).